cmd/root.go

```diff
@@ -315,6 +315,26 @@ func Execute() {
 		}
 	}
 
+	// Set CommandPrefix
+	scmd, _, _ := cmd.Find(os.Args)
+	for _, command := range RootCmd.Commands() {
+		if CommandPrefix != "" {
+			break
+		}
+		commandName := strings.Split(command.Use, " ")[0]
+		if scmd == command {
+			CommandPrefix = commandName + "."
+			break
+		}
+		for _, subcommand := range command.Commands() {
+			subCommandName := commandName + "_" + strings.Split(subcommand.Use, " ")[0]
+			if scmd == subcommand {
+				CommandPrefix = subCommandName + "."
+				break
+			}
+		}
+	}
```

Comment:
This looks involved enough to merit breaking into a function and unittesting specifically.

Reply:
I've been thinking about this comment for a bit.  The problem with unit testing is figuring out a way to fake a Command and subCommand structure and confirming the values that come out of it are correct.  I wonder if a better way is to add something like

if CommandPrefix != "mr_show." {
     log.Fatal("CommandPrefix", CommandPrefix", does not match "mr_show.")
}

throughout the test files?  That's a pain to maintain though.

Reply:
I'll definitely move this into a function.

Reply:
You don't wanna do a `log.Fatal` in the tests, to say the least, as it'll kill coverage output. There's `t.FailNow()` IIRC which if it doesn't accept a log statement can be preceded by `t.Log()`. Yeah, agreed tho, recreating the command structure is a bit annoying and a unittest won't catch if things move around inside cobra. 

Looks like you've already broken it out into a function now, so I'll call this thread resolved. I'm not concerned so much with unittesting as you've covered the functionality with the integration tests.